components/arm/arm_test.go

```diff
@@ -517,3 +517,273 @@ func (m *mockLocal) Close(ctx context.Context) error { m.reconfCount++; return n
+		checkMap := make(map[string]r3.Vector)
+		checkMap["rdk:component:arm:base_top"] = r3.Vector{
+			0.000000000000000000000000,
+			0.000000000000000000000000,
+			160.000000000000000000000000,
+		}
+		checkMap["rdk:component:arm:upper_arm"] = r3.Vector{
+			-26.820359657333263214695762,
+			0.000000000000000000000000,
+			399.308988008567553151806351,
+		}
+		checkMap["rdk:component:arm:upper_forearm"] = r3.Vector{
+			60.777555075062835499011271,
+			0.000000000000000000000000,
+			530.142781900699674224597402,
+		}
+		checkMap["rdk:component:arm:lower_forearm"] = r3.Vector{
+			175.357954329185588449036004,
+			0.000000000000000363462780,
+			331.043246286488795249169925,
+		}
+		checkMap["rdk:component:arm:wrist_link"] = r3.Vector{
+			297.258065257027055849903263,
+			27.068045067389423508075197,
+			256.363984524505951867467957,
+		}
+
+		in := []referenceframe.Input{{Value: 0}, {Value: -0.2}, {Value: -0.2}, {Value: -0.2}, {Value: -0.2}, {Value: -0.2}}
+		geoms, _ := notReal.ModelFrame().Geometries(in)
+		geomMap := geoms.Geometries()
+		b, outMap := locationCheckTestHelper(geomMap, checkMap)
+		test.That(t, b, test.ShouldBeTrue)
+		test.That(t, outMap, test.ShouldBeEmpty)
+	})
+}
+
+func TestUR5ELocations(t *testing.T) {
+	// check the exact values/locations of arm geometries at a couple different poses
+	logger := golog.NewTestLogger(t)
+	cfg := config.Component{
+		Name:  arm.Subtype.String(),
+		Model: resource.NewDefaultModel("fake"),
+		ConvertedAttributes: &fake.AttrConfig{
+			ArmModel: "ur5e",
+		},
+	}
+	notReal, err := fake.NewArm(cfg, logger)
+	test.That(t, err, test.ShouldBeNil)
+
+	t.Run("location check1", func(t *testing.T) {
+		checkMap := make(map[string]r3.Vector)
+		checkMap["rdk:component:arm:wrist_1_link"] = r3.Vector{
+			-817.200000000000045474735089,
+			-66.649999999999948840923025,
+			162.500000000000000000000000,
+		}
+		checkMap["rdk:component:arm:wrist_2_link"] = r3.Vector{
+			-817.200000000000045474735089,
+			-133.300000000000011368683772,
+			112.650000000000005684341886,
+		}
+		checkMap["rdk:component:arm:ee_link"] = r3.Vector{
+			-817.200000000000045474735089,
+			-183.149999999999920419213595,
+			62.799999999999940314410196,
+		}
+		checkMap["rdk:component:arm:base_link"] = r3.Vector{
+			0.000000000000000000000000,
+			0.000000000000000000000000,
+			120.000000000000000000000000,
+		}
+		checkMap["rdk:component:arm:upper_arm_link"] = r3.Vector{
+			-212.500000000000000000000000,
+			-130.000000000000000000000000,
+			162.499999999999971578290570,
+		}
+		checkMap["rdk:component:arm:forearm_link"] = r3.Vector{
+			-621.100000000000022737367544,
+			0.000000000000000000000000,
+			162.500000000000000000000000,
+		}
+
+		in := make([]referenceframe.Input, 6)
+		geoms, _ := notReal.ModelFrame().Geometries(in)
+		geomMap := geoms.Geometries()
+		b, outMap := locationCheckTestHelper(geomMap, checkMap)
+		test.That(t, b, test.ShouldBeTrue)
+		test.That(t, outMap, test.ShouldBeEmpty)
+	})
+	//nolint:dupl
```

Comment:
the linter was complaining about this. Specifically, it was saying that lines 695-734 are a duplicate of 736-775. However, this is not the case. confusing behavior.